File: WIP - Progetto/app/src/main/java/it/wip/database/dao/ChapterDao.kt
package it.wip.database.dao

import androidx.room.*
import it.wip.database.model.Chapter

@Dao
interface ChapterDao {

    @Query("SELECT * FROM chapter")
    suspend fun getAll(): Array<Chapter>

    @Query("SELECT * FROM chapter WHERE story = :chapterId")
    fun getAllById(chapterId: Int): Array<Chapter>

    @Query("SELECT * FROM chapter WHERE story = :storyId")
    fun getAllByStory(storyId: Int): Array<Chapter>

    @Insert
    suspend fun insert(vararg chapters: Chapter)

    @Insert
    fun insertWithoutCoroutines(vararg chapters: Chapter)

    @Update
    suspend fun update(chapter: Chapter)

    @Delete
    suspend fun delete(chapter: Chapter)

}

File: wip_flutter/lib/database/dao/chapter_dao.dart
import 'package:sqflite/sqflite.dart';

import '../model/chapter.dart';

class ChapterDao {

  static Future<List<Chapter>> getAll(Database db) async {
    final List<Map<String, dynamic>> maps = await db.query('chapter');
    return List.generate(maps.length, (i) {
      return Chapter(
          id: maps[i]['id'],
          chapterName: maps[i]['chapter_name'],
          time: maps[i]['time'],
          createdOn: maps[i]['created_on'],
          avatar: maps[i]['avatar'],
          story: maps[i]['story']
      );
    });
  }

  static Future<List<Chapter>> getAllByStory(Database db, int storyId) async {

    final List<Map<String, dynamic>> maps = await db.query(
        'chapter',
        columns: ['*'],
        where: 'story = ?',
        whereArgs: [storyId],
    );

    return List.generate(maps.length, (i) {
      return Chapter(
          id: maps[i]['id'],
          chapterName: maps[i]['chapter_name'],
          time: maps[i]['time'],
          createdOn: maps[i]['created_on'],
          avatar: maps[i]['avatar'],
          story: maps[i]['story']
      );
    });
  }

  static Future<void> insert(Database db, Chapter chapter) async {
    await db.insert('chapter', chapter.toMap());
  }

  static Future<void> insertAll(Database db, List<Chapter> chapters) async {
    for(Chapter chapter in chapters) {
      await db.insert('chapter', chapter.toMap());
    }
  }

  static Future<void> update(Database db, Chapter chapter) async {
    await db.update('chapter', chapter.toMap(), where: 'id = ?', whereArgs: [chapter.id]);
  }

  static Future<void> delete(Database db, int id) async {
    await db.delete('chapter', where: 'id = ?', whereArgs: [id]);
  }

}